src/sentry/testutils/cases.py

```diff
@@ -685,3 +689,78 @@ def setUp(self):
 
     def assertDialogSuccess(self, resp):
         assert 'window.opener.postMessage(' in resp.content
+
+
+class SnubaTestCase(TestCase):
+    def setUp(self):
+        super(SnubaTestCase, self).setUp()
+
+        assert requests.post(snuba.SNUBA + '/tests/drop').status_code == 200
+
+    def __wrap_event(self, event, data, primary_hash):
+        # TODO: Abstract and combine this with the stream code in
+        #       getsentry once it is merged, so that we don't alter one
+        #       without updating the other.
+        return {
+            'group_id': event.group_id,
+            'event_id': event.event_id,
+            'project_id': event.project_id,
+            'message': event.message,
+            'platform': event.platform,
+            'datetime': event.datetime,
+            'data': data,
+            'primary_hash': primary_hash,
+        }
+
+    def create_event(self, *args, **kwargs):
+        """\
+        Takes the results from the existing `create_event` method and
+        inserts into the local test Snuba cluster so that tests can be
+        run against the same event data.
+
+        Note that we create a GroupHash as necessary because `create_event`
+        doesn't run them through the 'real' event pipeline. In a perfect
+        world all test events would go through the full regular pipeline.
+        """
+
+        event = super(SnubaTestCase, self).create_event(*args, **kwargs)
+
+        data = event.data.data
+        tags = dict(data.get('tags', []))
+
+        if not data.get('received'):
+            data['received'] = calendar.timegm(event.datetime.timetuple())
+
+        environment = Environment.get_or_create(
+            event.project,
+            tags['environment'],
+        )
+
+        GroupEnvironment.objects.get_or_create(
+            environment_id=environment.id,
+            group_id=event.group_id,
+        )
+
+        try:
+            grouphash = GroupHash.objects.filter(group=event.group).order_by('hash')[0]
+        except IndexError:
+            grouphash = GroupHash.objects.create(
+                project=event.project,
+                group=event.group,
+                hash=uuid.uuid4().hex,
+            )
```

Comment:
This doesn't necessarily give you the primary hash (or it can create a hash that's totally disassociated with the event?) If you want the primary hash, you can do:

```python
from sentry.event_manager import get_hashes_from_fingerprint, md5_from_hash

primary_hash = md5_from_hash(
    get_hashes_from_fingerprint(
        event,
        data.get('fingerprint', ['{{ default }}']),
    )[0],
)
```

Reply:
Ya, that was a total hack because I didn't realize it was so easy to get the real one, nice.

Reply:
Yeah, we should probably just write something to return that since I think we've implemented it (or some similar variation) a few times. 😬